src/automata/FireAndForgetRequester.h

```diff
@@ -0,0 +1,55 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#pragma once
+
+#include <memory>
+
+#include "src/Payload.h"
+
+#include "src/ReactiveStreamsCompat.h"
+#include "src/mixins/ConsumerMixin.h"
+#include "src/mixins/ExecutorMixin.h"
+#include "src/mixins/LoggingMixin.h"
+#include "src/mixins/MemoryMixin.h"
+#include "src/mixins/MixinTerminator.h"
+#include "src/mixins/PublisherMixin.h"
+#include "src/mixins/SinkIfMixin.h"
+#include "src/mixins/SourceIfMixin.h"
+#include "src/mixins/StreamIfMixin.h"
+
+namespace reactivesocket {
+
+/// Implementation of an automaton that represents a Fire-and-Forget requester.
+/// Note that it's simpler than other automata due to the inherent throwaway
+/// simplicity of Fire-and-Forget semantics. Stream IDs are not tracked.
+///
+/// TODO: we need to add logging similar to other *Requester automata.
+class FireAndForgetRequester: public Subscription {
```

Comment:
> and the client calling request(N)

It's not clear to me why would the client call it. What does that mean, if we never deliver anything on that subscription?

Why are you concerned by ref counting? It's thread-unsafe, hence as fast as any other implementation of memory management over here.